src/algebra/big_operators/basic.lean

```diff
@@ -1718,6 +1718,48 @@ begin
   { rw [finset.sum_insert hat, finset.prod_insert hat, multiset.prod_add, ih] }
 end
 
+lemma disjoint_of_disjoint_sum {α β : Type*} {i : finset β} {f : β → multiset α} {z : β}
+  (h : ∀ x ∈ i, multiset.disjoint (f x) (f z)) :
+  multiset.disjoint (f z) (∑ x in i, f x) :=
+begin
+  classical,
+  apply finset.sum_induction f (λ (t : multiset α), disjoint (f z) t),
+  { exact λ _ _ ha hb, disjoint_add_right.mpr ⟨ha, hb⟩ },
+  { exact (zero_disjoint (f z)).symm},
+  { exact λ x hx, (h x hx).symm },
+end
```

Comment:
It would be good to have the `list` version too:
```lean
lemma disjoint_list_sum_right {α : Type*} {a : multiset α} {l : list (multiset α)}
  (h : ∀ b ∈ l, multiset.disjoint a b) :
  multiset.disjoint a l.sum :=
begin
  induction l with b bs ih,
  { exact (zero_disjoint _).symm },
  { rw [list.sum_cons, disjoint_add_right],
    exact ⟨h _ (list.mem_cons_self _ _), ih $ λ b hb, h _ $ list.mem_cons_of_mem _ hb⟩ }
end

lemma disjoint_sum_right {α : Type*} {a : multiset α} {i : multiset (multiset α)} :
  (∀ b ∈ i, multiset.disjoint a b) → multiset.disjoint a i.sum :=
quotient.induction_on i $ λ l h, begin
  rw [quot_mk_to_coe, multiset.coe_sum],
  exact disjoint_list_sum_right h,
end

lemma disjoint_finset_sum_right {α β : Type*} {i : finset β} {f : β → multiset α} {a : β}
  (h : ∀ b ∈ i, multiset.disjoint (f a) (f b)) :
  multiset.disjoint (f a) (∑ x in i, f x) :=
disjoint_sum_right $ by simpa using h
```

Reply:
Ok. Thanks, I am adding these results